File: core/src/main/java/com/ustadmobile/core/view/GroupDetailView.java
package com.ustadmobile.core.view;


import com.ustadmobile.core.db.UmProvider;
import com.ustadmobile.lib.db.entities.PersonGroup;
import com.ustadmobile.lib.db.entities.PersonWithEnrollment;

/**
 * Core View. Screen is for GroupDetail's View
 */
public interface GroupDetailView extends UstadView {


    // This defines the view name that is an argument value in the go() in impl.
    String VIEW_NAME = "GroupDetail";

    //Any argument keys:
    String GROUP_UID = "GroupUid";

    void setListProvider(UmProvider<PersonWithEnrollment> provider);

    void updateGroupOnView(PersonGroup group);

    /**
     * Method to finish the screen / view.
     */
    void finish();


}

File: core/src/commonMain/kotlin/com/ustadmobile/staging/core/view/GroupDetailView.kt
package com.ustadmobile.core.view


import androidx.paging.DataSource
import com.ustadmobile.core.db.UmProvider
import com.ustadmobile.lib.db.entities.PersonGroup
import com.ustadmobile.lib.db.entities.PersonWithEnrollment

/**
 * Core View. Screen is for GroupDetail's View
 */
interface GroupDetailView : UstadView {

    fun setListProvider(provider: DataSource.Factory<Int, PersonWithEnrollment>)

    fun updateGroupOnView(group: PersonGroup)

    /**
     * Method to finish the screen / view.
     */
    fun finish()

    companion object {


        // This defines the view name that is an argument value in the go() in impl.
        val VIEW_NAME = "GroupDetail"

        //Any argument keys:
        val GROUP_UID = "GroupUid"
    }


}

